### PR TITLE
fix embarassing colour error in the nightly status button

### PR DESCRIPTION
**Pull request by oleorhagen**
Signed-off-by: Ole Petter <ole.orhagen@northern.tech>

### CODE REVIEW

File: ui/pages/build-status.js [hunk 1]
```diff
@@ -88,6 +88,22 @@ const RepoStatusItem = ({ repo, organization = 'Mender', branch = 'master', cove
   </Stack>
 );
 
+const buildStatusColor = (status) => {
+    switch(status) {
+    case 'FAILED':
+        return 'error';
+        break;
+    case 'RUNNING':
+        return 'inprogress';
```
Comment (mzedel) [inline ui/pages/build-status.js:97]: `inprogress` doesn't seem to be part of the [mui palette](https://mui.com/material-ui/customization/palette/)

File: ui/pages/build-status.js [hunk 2]
```diff
@@ -88,6 +88,22 @@ const RepoStatusItem = ({ repo, organization = 'Mender', branch = 'master', cove
...
+        break;
+    case 'SUCCESS':
+        return 'success';
+        break;
+    default:
+        return '#FFA500'; // Orange
```
Comment (mzedel) [inline ui/pages/build-status.js:103]: to make use of the theming capabilities I would suggest to use `warning` from [here](https://mui.com/material-ui/customization/palette/) or define a color in the theme 👍

File: ui/pages/build-status.js [hunk 3]
```diff
@@ -88,6 +88,22 @@ const RepoStatusItem = ({ repo, organization = 'Mender', branch = 'master', cove
...
+    }
+}
```
Comment (mzedel) [inline ui/pages/build-status.js:105]:
```suggestion
const buildStatusColorMap = {
    FAILED: 'error',
    RUNNING: 'inprogress',
    SUCCESS: 'success',
    default: 'warning'
}
```
and then use it with `buildStatusColorMap[latestNightly.status] || buildStatusColorMap.default` or `buildStatusColorMap[latestNightly.status] ?? buildStatusColorMap.default`

**Review by mzedel — CHANGES_REQUESTED** [review]
👍 🚀

File: ui/pages/build-status.js (hunk 3 above)
Comment (oleorhagen) [reply ui/pages/build-status.js:105]: Oh my, whole new world :heart_eyes_cat:

File: ui/pages/build-status.js [hunk 4]
```diff
@@ -88,6 +88,16 @@ const RepoStatusItem = ({ repo, organization = 'Mender', branch = 'master', cove
...
+const buildStatusColor = (status) => {
+    const buildStatusColorMap = {
+        'FAILED': 'error',
+        'RUNNING': 'warning',
+        'SUCCESS': 'success',
+        default: 'warning.dark' // WTF is going on colour!
+    };
+    return buildStatusColorMap[status] || buildStatusColorMap.default;
+};
+
```
Comment (mzedel) [inline ui/pages/build-status.js]:
```suggestion
    const buildStatusColorMap = {
        FAILED: 'error',
        RUNNING: 'warning',
        SUCCESS: 'success',
        default: 'warning.dark' // WTF is going on colour!
    };
```

File: ui/pages/build-status.js [hunk 5]
```diff
@@ -102,7 +112,7 @@ const BuildStatus = ({ componentsByArea, latestNightly, ltsReleases, versions })
             variant="outlined"
             title={latestNightly.startedAt}
             onClick={openNightlyClick}
-            endIcon={<Circle color={latestNightly.status === 'FAILED' ? 'error' : 'success'} />}
+            endIcon={<Circle color={buildStatusColor(latestNightly.status)} />}
```
Comment (mzedel) [inline ui/pages/build-status.js:115]:
```suggestion
            endIcon={<Circle color={buildStatusColorMap[latestNightly.status] || buildStatusColorMap.default} />}
```

File: ui/pages/build-status.js [hunk 6]
```diff
@@ -88,6 +88,16 @@ const RepoStatusItem = ({ repo, organization = 'Mender', branch = 'master', cove
...
 
+const buildStatusColor = status => {
+  const buildStatusColorMap = {
+    'FAILED': 'error',
+    'RUNNING': 'warning',
+    'SUCCESS': 'success',
+    default: 'warning.dark' // WTF is going on colour!
+  };
+  return buildStatusColorMap[status] || buildStatusColorMap.default;
+};
```
Comment (mzedel) [inline ui/pages/build-status.js:99]:
```suggestion
const buildStatusColorMap = {
    FAILED: 'error',
    RUNNING: 'warning',
    SUCCESS: 'success',
    default: 'warning.dark' // WTF is going on colour!
};
```
mind the lack of `'` for the uppercase entries... consistency! (unless of course we enter `'09-FAILED-ONLY-PARTLY-%': 'grey'` into the map)

**Review by mzedel — CHANGES_REQUESTED** [review]
not that this would be in any way a crucial and frequently rendered part, but we can have the look up work most of the time and omit the step of calling a function 100% of the time...

**Review by mzedel — APPROVED** [review]
😻

**Pull request merged**